record_tx_strategy_outgoing_tx.go

```diff
@@ -82,7 +82,12 @@ func (strategy *outgoingTx) Validate() error {
 }
 
 func (strategy *outgoingTx) TxID() string {
-	btTx, _ := bt.NewTxFromString(strategy.Hex)
+	btTx, err := bt.NewTxFromString(strategy.Hex)
+
+	// Return hex if hex is invalid. Handle error in error handlers
+	if err != nil {
```

Comment:
I recall we discussed about it earlier. But now, I think that, maybe it's better to validate the Hex in the Validate method which should be called before `Execute`'ing the strategy. 

Next idea is: instead of weird `go-bt` package panic, raise our own explicit panic. 

What do you think?

This is what I'm thinking about. It's only a suggestion - I haven't test it locally.
```golang
func (strategy *outgoingTx) Validate() error {
	if strategy.Hex == "" {
		return ErrMissingFieldHex
	}

   //additional validation of the Hex to make sure that when we call TxID we won't get panic
	if _, err := bt.NewTxFromString(strategy.Hex); err != nil {
		return fmt.Errorf("invalid hex: %w", err)
	}

	if strategy.RelatedDraftID == "" {
		return errors.New("empty RelatedDraftID")
	}

	if strategy.XPubKey == "" {
		return errors.New("empty xPubKey")
	}

	return nil // is valid
}

// TxID returns the transaction ID based of correct HEX; the transaction along with the HEX value should be validated first
func (strategy *outgoingTx) TxID() string {
	btTx, err := bt.NewTxFromString(strategy.Hex)
	if err != nil {
        //on wrong/nil btTx the panic will raise nonetheless - so it's better to raise an explicit error here
		panic(errors.New("invalid hex - TxID was called before validation"))
	}
	return btTx.TxID()
}
```

Reply:
`TxID` should return valid ID and only ID of transaction (with err if needed). Please check this in `Validate()` function. 

Current solution may cause problems with understanding the code in the future

Reply:
The problem is, that `Validate` method is not always called before using the `TxID` method. So, the idea, "to validate before..." includes broader perspective and additional changes. 

Maybe would be enough to modify the `recordTxStrategy` that `LockKey` and `TxID` methods, both return additional error as a second value:

```golang
type recordTxStrategy interface {
	TxID() (string, error)
	LockKey() (string, error)
	Validate() error
	Execute(ctx context.Context, c ClientInterface, opts []ModelOps) (*Transaction, error)
}
``` 
Of course this modification would require changes in all structs which implement this interface:
- externalIncomingTx
- internalIncomingTx
- outgoingTx

Reply:
So, my idea was to write code where, when the strategy is in use, all the data it possesses is valid, aiming to avoid if err= statements. However, as @chris-4chain pointed out, there is currently no mechanism to enforce the use of Validate(). I overlooked calling it for incoming paymail transactions.

In my opinion, a quick win would be to invoke this validation before using the strategy in the paymail logic.

In ideal world we could force using proper factory function by hiding some logic in packages... it's impossible now because we have everything in one 'bux' package